util/pkg/vfs/s3context.go

```diff
@@ -34,6 +35,8 @@ type S3Context struct {
 	mutex           sync.Mutex
 	clients         map[string]*s3.S3
 	bucketLocations map[string]string
+
+	getClient func(region string) (*s3.S3, error)
```

Comment:
My personal choice would be not to do this, and just to have an `if` statement in getClient that calls `getCustomClient` or similar, but this is fine if you prefer it.